[WIP] Fusillade Process Management for DCP

Bento007 opened the pull request.

Bento007 committed:
WIP

DailyDreaming commented on rfcs/text/Fusillade in the DCP.md:
```diff
@@ -0,0 +1,105 @@
+
+1. Create a file **{role}.json** for each user type. Using the resource and actions previously define, write a policy in the AWS IAM policy format that express what each role can do. If it isn't explicitly stated, it is assumed the user with this role does not have permission. For the resource field, `*` can be used as a wild card in the same way it's used in AWS IAM policies.
+
+1. Run the script `??? -path "path to roles" -component "name of the component" -stage "current deployment stage" -secret "secretId of a service account in AWS" -service "path to service account credentials"` to add your roles to fusillade. The role that is added to fusillade will be named `{component}-{stage}-{role}`. The stage is not included for *prod* deployments.
```
`??? -path "path to roles" -component "name of the component" -stage "current deployment stage" -secret "secretId of a service account in AWS" -service "path to service account credentials"`

Guessing this should include a `-role "role_name"` option if it's going to be named `{component}-{stage}-{role}`.  I think it would be useful to warn users when a role with that name already exists, and maybe add a `--force` option to overwrite existing roles.  Also, kind of a nit, but I think I prefer `--` options rather than `-` just to be consistent with the `data-store`.

Bento007 replied:
The path would be to a folder which contains all of the roles, so they can be updated all at once. It's probably better to do it one at a time so you know what role failed. I'll add `--` to the parameters, `--role`, change `--path` to `--file`. Error returned if role exists, and `--force` will overwrite the role if the roles exists.

DailyDreaming commented on rfcs/text/Fusillade in the DCP.md:
```diff
@@ -0,0 +1,105 @@
+service account in AWS" -service "path to service account credentials"` to add your group to Fusillade. If any roles 
+specified in the group do not exists, an error will ne thrown and that group will not be create if it didn't exist, 
+or it will not be updated if it existed. Any member not in the group will be added to the group. The group added will
+ be named `{group}-{deployment}`. `{deployment}` is excluded from name when deployment is `prod`. For all deployments
```
`For all deployments the group named user_default does is not contain the deployment.` was unclear to me.

Bento007 replied:
fixed

DailyDreaming commented on rfcs/text/Fusillade in the DCP.md:
````diff
@@ -0,0 +1,105 @@
+}
+``` 
+
+1. In DCP infra run the script `??? -path "path to groups" -stage "current deployment stage" -secret "secretId of a 
````
Same as the above, I'm assuming there needs to be an argument for the unique group name and a check.

Bento007 replied:
fixed

DailyDreaming commented on rfcs/text/Fusillade in the DCP.md:
```diff
@@ -0,0 +1,105 @@
+## Detailed Design
+
+### As a component of the DCP, how do I control access to my resources?
+1. Define what actions and resources you'd like to control. Actions should take the form `{component}:{Action}`, for example to get a bundle from the DSS the action could be `dss:GetBundle`. The action is case sensitive, and camel case is suggested. Resources are what actions affect. The same action can be applied to multiple different resources. For the resource use the AWS ARN format. For examp ...
```
Should there be a distinction between `dss:GetBundle` for AWS and GCP?

Bento007 replied:
My opinion is that this would be a field in the resource.

Bento007 replied:
Components have the freedom to determine the relationship between actions and resources. I will expand on the guide lines for defining resources, and add some examples.

Bento007 replied:
I added additional documentation for resources to help make this clear for components

Bento007 committed:
Adding more clarification and links.

- added directory structure for components and dcpinfra
- added more examples
- moved cli tools to it's own section
- added more link to fusillade resources
- added additional how tos

Bento007 committed:
Grammer and spelling fixes

Bento007 committed:
adding more thoughts.
